cl/phase1/network/services/attestation_service.go

```diff
@@ -186,56 +299,77 @@ func (s *attestationService) ProcessMessage(ctx context.Context, subnet *uint64,
 	if err != nil {
 		return fmt.Errorf("unable to get the domain: %v", err)
 	}
-	signingRoot, err := computeSigningRoot(att.AttestantionData(), domain)
+	signingRoot, err := computeSigningRoot(att.Attestation.AttestantionData(), domain)
 	if err != nil {
 		return fmt.Errorf("unable to get signing root: %v", err)
 	}
-	if valid, err := blsVerify(signature[:], signingRoot[:], pubKey[:]); err != nil {
-		return err
-	} else if !valid {
-		log.Warn("invalid signature", "signature", common.Bytes2Hex(signature[:]), "signningRoot", common.Bytes2Hex(signingRoot[:]), "pubKey", common.Bytes2Hex(pubKey[:]))
-		return errors.New("invalid signature")
-	}
+	// if valid, err := blsVerify(signature[:], signingRoot[:], pubKey[:]); err != nil {
+	// 	return err
+	// } else if !valid {
+	// 	log.Warn("invalid signature", "signature", common.Bytes2Hex(signature[:]), "signningRoot", common.Bytes2Hex(signingRoot[:]), "pubKey", common.Bytes2Hex(pubKey[:]))
+	// 	return errors.New("invalid signature")
+	// }
 
-	// [IGNORE] The block being voted for (attestation.data.beacon_block_root) has been seen (via both gossip and non-gossip sources)
-	// (a client MAY queue attestations for processing once block is retrieved).
-	if _, ok := s.forkchoiceStore.GetHeader(root); !ok {
-		s.scheduleAttestationForLaterProcessing(att)
-		return ErrIgnore
+	aggregateVerificationData := &AggregateVerificationData{
+		Signatures: [][]byte{signature[:]},
+		SignRoots:  [][]byte{signingRoot[:]},
+		Pks:        [][]byte{pubKey[:]},
+		GossipData: att.GossipData,
 	}
 
-	// [REJECT] The attestation's target block is an ancestor of the block named in the LMD vote -- i.e.
-	// get_checkpoint_block(store, attestation.data.beacon_block_root, attestation.data.target.epoch) == attestation.data.target.root
-	startSlotAtEpoch := targetEpoch * s.beaconCfg.SlotsPerEpoch
-	if targetBlock := s.forkchoiceStore.Ancestor(root, startSlotAtEpoch); targetBlock != att.AttestantionData().Target().BlockRoot() {
-		return fmt.Errorf("invalid target block. root %v targetEpoch %v attTargetBlockRoot %v targetBlock %v", root.Hex(), targetEpoch, att.AttestantionData().Target().BlockRoot().Hex(), targetBlock.Hex())
-	}
-	// [IGNORE] The current finalized_checkpoint is an ancestor of the block defined by attestation.data.beacon_block_root --
-	// i.e. get_checkpoint_block(store, attestation.data.beacon_block_root, store.finalized_checkpoint.epoch) == store.finalized_checkpoint.root
-	startSlotAtEpoch = s.forkchoiceStore.FinalizedCheckpoint().Epoch() * s.beaconCfg.SlotsPerEpoch
-	if s.forkchoiceStore.Ancestor(root, startSlotAtEpoch) != s.forkchoiceStore.FinalizedCheckpoint().BlockRoot() {
-		return fmt.Errorf("invalid finalized checkpoint %w", ErrIgnore)
```

Comment:
why are you not returning error anymore

Reply:
All of these checks where after signature verification. That's why I left them that way. So you saying it's ok to run those checks and return err BEFORE signature verification?

Reply:
no, move them before